### DISCUSSION TITLE
How to read vocabulary file from s3?

### DISCUSSION THREAD

**Discussion by adarshghagta** [deepjavalibrary/djl#1992] [Q&A]
Can addFromTextFile() in DefaultVocabulary class work with an S3 url as input?

**Comment by zachgk** [comment]
No, it is implemented with the Java URL API so it shouldn't. Instead, you could download from S3 to a temp file and then load the temp file using `addFromTextFile()`. You can use the Amazon SDK package `software.amazon.awssdk:s3` to retrieve the file. Alternatively, you could also just add the words to the DefaultVocabulary directly
